stock_picker.rb

```diff
@@ -0,0 +1,52 @@
+def stock_picker(prices_array)
+  highest_profit = -10**6
+  highest_profit_interval = []
+
+  prices_array.each_with_index do |buying_price, buy_day_index|
+    break if buy_day_index == prices_array.size - 1
+
+    prices_after_buy_day_index = prices_array[buy_day_index + 1..]
+    max_selling_price_for_buy_day = prices_after_buy_day_index.max
+    max_profit_for_buy_day = max_selling_price_for_buy_day - buying_price
+    next unless max_profit_for_buy_day > highest_profit
+
+    highest_profit = max_profit_for_buy_day
+    highest_profit_interval = [buy_day_index,
+                               prices_after_buy_day_index.index(max_selling_price_for_buy_day) + buy_day_index + 1]
+  end
+  highest_profit_interval
+end
+
+def stock_picker_combination(prices_array)
+  (0..prices_array.length - 1).to_a.combination(2).max_by do |pair|
+    (prices_array[pair[1]]) - (prices_array[pair[0]])
+  end
+end
+
+def stock_picker_with_min(prices_array)
+  min = 10**6
+  highest_profit = -10**6
+  highest_interval = []
+  min_index = 0
+  prices_array.each_with_index do |price, price_index|
```

Comment:
just a suggestion - but IMO the code can be more readable by setting initial values from the array

```suggestion
def stock_picker_with_min(prices_array)
  raise "need at least 2 days" if array.length < 2
  
  min = prices_array[0..1].min
  highest_profit = prices_array[1] - prices_array[0]
  highest_interval = [0,1]
  min_index = prices_array[0..1].index(min)
  prices_array[2..].each_with_index do |price, price_index|
```

WDYT?